autoPyTorch/pipeline/components/setup/network_backbone/ShapedResNetBackbone.py

```diff
@@ -38,10 +38,14 @@ def build_backbone(self, input_shape: Tuple[int, ...]) -> None:
         self.config.update(
             {"num_units_%d" % (i): num for i, num in enumerate(neuron_counts)}
         )
-        if self.config['use_dropout'] and self.config["max_dropout"] > 0.05:
+        # we are skipping the last layer, as the function get_shaped_neuron_counts
+        # is built for getting neuron counts, so it will add the out_features to
+        # the last layer. However, in dropout we dont want to have that, we just
+        # want to use the shape and not worry about the output.
+        if self.config['use_dropout']:
             dropout_shape = get_shaped_neuron_counts(
-                self.config['resnet_shape'], 0, 0, 1000, self.config['num_groups']
-            )
+                self.config['resnet_shape'], 0, 0, 1000, self.config['num_groups'] + 1
+            )[:-1]
```

Comment:
Hi, I immediately do not understand this.  The comment is very clear and I get the intention... Can we add a unit test for this? 

So that we call `build_backbone` and we get a model. Then we check that dropout was added as expected?

Reply:
sure